automapper.go

```diff
@@ -30,13 +35,31 @@ import (
 // destination to ensure that a renamed field in either the source or
 // destination does not result in subtle silent bug.
 func Map(source, dest interface{}) {
+	MapWithOptions(source, dest, MapOptions{})
+}
+
+// MapWithOptions fills out the fields in dest with values from source. All fields in the
+// destination object must exist in the source object.
+//
+// Object hierarchies with nested structs and slices are supported, as long as
+// type types of nested structs/slices follow the same rules, i.e. all fields
+// in destination structs must be found on the source struct.
+//
+// Embedded/anonymous structs are supported
+//
+// Values that are not exported/not public will not be mapped.
+//
+// It is a design decision to panic when a field cannot be mapped in the
+// destination to ensure that a renamed field in either the source or
+// destination does not result in subtle silent bug.
+func MapWithOptions(source, dest interface{}, opts MapOptions) {
```

Comment:
I think the use of an options struct as parameter for a function is accepted within the Go community. So this signature looks fine to me

Reply:
Too bad that Go doesn't support overloads, or optional parameters. Would prefer to keep the method name the same with an overload.